src/serializers/HdfSerializer.cpp

```diff
@@ -0,0 +1,127 @@
+/********************************************************************************
+ *                                                                              *
+ * This file is part of IfcOpenShell.                                           *
+ *                                                                              *
+ * IfcOpenShell is free software: you can redistribute it and/or modify         *
+ * it under the terms of the Lesser GNU General Public License as published by  *
+ * the Free Software Foundation, either version 3.0 of the License, or          *
+ * (at your option) any later version.                                          *
+ *                                                                              *
+ * IfcOpenShell is distributed in the hope that it will be useful,              *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 *
+ * Lesser GNU General Public License for more details.                          *
+ *                                                                              *
+ * You should have received a copy of the Lesser GNU General Public License     *
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.         *
+ *                                                                              *
+ ********************************************************************************/
+
+
+#include "HdfSerializer.h"
+
+#include "../ifcgeom_schema_agnostic/IfcGeomRenderStyles.h"
+
+#include "../ifcparse/utils.h"
+
+#include <boost/lexical_cast.hpp>
+#include <iomanip>
+
+
+
+HdfSerializer::HdfSerializer(const std::string& hdf_filename, const SerializerSettings& settings)
+	: GeometrySerializer(settings)
+	, hdf_filename(hdf_filename)
+	, DATASET_NAME_POSITIONS("Positions")
+	, DATASET_NAME_NORMALS("Normals")
+	, DATASET_NAME_INDICES("Indices")
+	, DATASET_NAME_OCCT("OCCT Text")
+
+{
+	guids = {};
+}
+
+
+bool HdfSerializer::ready() {
+	//todo: check whether the file exists
+	return true;
+}
+
+void HdfSerializer::writeHeader() {
+	const H5std_string  FILE_NAME(hdf_filename);
+	file = H5::H5File(FILE_NAME, H5F_ACC_TRUNC);
+
+}
+
+
+void HdfSerializer::write(const IfcGeom::BRepElement<real_t>* o) {
+
+	std::string guid = o->guid();
+
+	H5::Group elementGroup;
+
+	H5::Group meshGroup;
+	H5::DataSet positionsDataset;
+	H5::DataSet normalsDataset;
+	H5::DataSet indicesDataset;
+
+	H5::Group OCCTGroup;
+	H5::DataSet OCCTDataset;
+	
+	const IfcGeom::Representation::BRep& brepmesh = o->geometry();
+	const IfcGeom::Representation::Serialization *serialization = new IfcGeom::Representation::Serialization(brepmesh);
```

Comment:
Ah one more thing, you create a new pointer (heap allocation) but you don't free (delete) the result.

In this particular case it'd be sufficient to do `const IfcGeom::Representation::Serialization serialization(brepmesh)`, meaning the element will be on the stack and automatically freed when it goes out of scope. Same for the triangulation element below.